src/comment.rs

```diff
@@ -240,6 +244,33 @@ pub enum CodeCharKind {
     Comment,
 }
 
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+enum FullCodeCharKind {
```

Comment:
Could you add a comment on this enum please? And maybe CodeCharKind too, explaining when you would use one or the other and what exactly the variants in Full... mean